microovn/frr/bgp/redirect.go

```diff
@@ -238,28 +284,157 @@ func redirectBgp(ctx context.Context, s state.State, extConnections []types.BgpE
 			"--",
 			"add-port", intBr, bgpIface,
 			"--",
+			"set", "Port", bgpIface, fmt.Sprintf("external-ids:%s=true", BgpManagedTag),
+			fmt.Sprintf("external-ids:%s=%s", BgpVrfTable, vrfName), fmt.Sprintf("external-ids:%s=%s", BgpIfaceIP, bgpIfaceIP4),
```

Comment:
Hmm, it may get bit complicated, but it should be possible. 

> We create and maintain this port as a side effect of there being a options:routing-protocol-redirect key present on the LRP whose value is the name of this port.

Value of this option is the LSP name, not the port. However what we could do, during the restart/reboot, we could iterate over all `Port`s to find those tagged with `BgpManagedTag` and for each:
* get associated `Interface`
* get `iface-id` from the `Interface` to identify LSP
* get LSP's peer to identify LRP
* get IP from LRP
* find LRP's LR
* get `requested-tnl-key` from LR

I'm bit on the fence whether it's worth it doing it this way. What do you think?

Reply:
This remains unchanged. The alternative approach seems bit too complicated to me. Especially given that we don't (currently) support reconfiguration, only enable/disable. So I don't see much opportunity for these values to get out of sync